### PR TITLE
flathub: Add org.caione.GScope

### PR DESCRIPTION
**Pull request by carlocaione**
gScope is source browsing tool for large and complex projects written in C.

### CODE REVIEW

File: org.caione.GScope.json [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+    ],
+    "modules": [
+        {
+            "name" : "gtksourceview",
+            "config-opts" : [ "--disable-Werror", "--disable-doc" ],
+            "sources" : [
+                {
+                    "type" : "git",
+                    "branch" : "gnome-3-24",
+                    "url" : "https://gitlab.gnome.org/GNOME/gtksourceview.git"
```
Comment (scx) [inline org.caione.GScope.json]: https://github.com/flathub/flathub/wiki/App-Requirements#stable-releases-reproducible-builds

Comment (carlocaione) [reply org.caione.GScope.json]: makes sense. Fixing.

File: org.caione.GScope.json [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+                    "type": "archive",
+                    "url": "https://sourceforge.net/projects/cscope/files/cscope/v15.9/cscope-15.9.tar.gz",
+                    "sha256": "c5505ae075a871a9cd8d9801859b0ff1c09782075df281c72c23e72115d9f159"
+                }
+            ]
+        },
+        {
+            "name": "gscope",
+            "buildsystem": "meson",
+            "config-opts": [ "--libdir=lib" ],
```
Comment (barthalion) [inline org.caione.GScope.json]: This and `builddir` are set by default for meson buildsystem.

File: org.caione.GScope.json [hunk 3]
```diff
@@ -0,0 +1,68 @@
+{
+    "app-id": "org.caione.GScope",
+    "runtime": "org.gnome.Platform",
+    "runtime-version": "3.30",
+    "sdk": "org.gnome.Sdk",
+    "command": "gscope",
+    "finish-args": [
+        "--share=network",
```
Comment (barthalion) [inline org.caione.GScope.json]: Which features require network access?

**Comment by barthalion** [comment]
bot, build org.caione.GScope

**Comment by barthalion** [comment]
I don't think it correctly uses the portal – clicking "Open" gives me empty listing instead of files in home directory.

**Comment by carlocaione** [comment]
> I don't think it correctly uses the portal – clicking "Open" gives me empty listing instead of files in home directory.

At least I get the content of my home directory. Let me check.

File: org.caione.GScope.json (hunk 1 above)
Comment (TingPing) [reply org.caione.GScope.json]: I'd suggest just using the tarballs release over tags, just smaller and faster downloads.

**Comment by nedrichards** [comment]
Same, it doesn't work for me either. I may have an older portals implementation though as I'm using Endless.

**Comment by carlocaione** [comment]
Uhm, I definitely get the content of my home directory. Could you check if adding `"--filesystem=host"` makes any difference?

Also not sure if using `GtkFileChooser` instead of `GtkFileChooserNative` can be the problem here?

**Comment by nedrichards** [comment]
Yes, that is entirely the problem. You can't get access to the portal using the normal filechooser, you have to use native (I know there's a plan for a future version of GTK+ to move that to be the default)

**Comment by carlocaione** [comment]
Ok, but `GtkFileChooserNative` doesn't support `GTK_FILE_CHOOSER_ACTION_SELECT_FOLDER` so AFAICT it falls back on the old `GtkFileChooser`. Is this a chicken-egg problem?

**Comment by barthalion** [comment]
Yes, given lack of directories support, it needs `--filesystem=home` at least.

**Comment by barthalion** [comment]
Any news? It's good to merge with `--filesystem=home` added.

**Comment by carlocaione** [comment]
@barthalion @nedrichards Punched a `"--filesystem=host` hole.

**Comment by barthalion** [comment]
bot, build org.caione.GScope

**Comment by barthalion** [comment]
Repository has been created: https://github.com/flathub/org.caione.GScope

Thanks!

**Pull request closed without merging**